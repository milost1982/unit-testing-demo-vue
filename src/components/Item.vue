<template>
  <div class="my-item">{{ `${value.id}: ${value.name}` }}</div>
</template>

<script>
export default {
  props: {
    value: Object,
    default: () => 'Default Item'
  }
}
</script>

<style>
.my-item {
  border: 1px dashed blue;
  margin: 10px;

}
.my-item:hover {
  background-color: lightblue;
}
</style>
