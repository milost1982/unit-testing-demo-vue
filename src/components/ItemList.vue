<template>
  <div class="item-list-component">
    <h2>List of items:</h2>
    <Item
      v-for="item in items"
      :key="item.id"
      :value="item"
    />
  </div>
</template>

<script>
import Item from './Item.vue'

export default {
  components: {
    Item
  },
  props: {
    items: {
      type: Array,
      default: () => []
    }
  }
}
</script>

<style scoped>
.item-list-component {
  width: 200px;
  margin: 0 auto;
}

</style>
