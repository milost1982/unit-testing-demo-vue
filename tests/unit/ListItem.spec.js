import { mount } from '@vue/test-utils'
import ItemList from '@/components/ItemList.vue'
import Item from '@/components/Item.vue'

describe('ListItem Component', () => {
  it('renders items passed as props', () => {

    // expect(true).toBe(false)
    const input = [
      { id: 10, name: 'Item 1' },
      { id: 20, name: 'Item 2' },
      { id: 30, name: 'Item 3' }
    ]

    const wrapper = mount(ItemList, {
      propsData: {
          items: input
        }
    })

    let itemWrappers = wrapper.findAllComponents(Item)
    console.log(JSON.stringify(itemWrappers.at(1).vm.value))

    expect(itemWrappers.length).toEqual(3)

    expect(itemWrappers.at(1).vm.value).toMatchObject(input[1])
    expect(wrapper).toMatchSnapshot()
  })
})
