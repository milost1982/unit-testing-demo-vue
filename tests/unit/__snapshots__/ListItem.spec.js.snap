// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Component renders items passed as props 1`] = `
<div class="item-list-component">
  <h2>List of items:</h2>
  <div class="my-item">10: Item 1</div>
  <div class="my-item">20: Item 2</div>
  <div class="my-item">30: Item 3</div>
</div>
`;
